src/AlgebraicGeometry/TropicalGeometry/groebner_fan.jl

```diff
@@ -0,0 +1,541 @@
+    ###
+    G = groebner_basis(I,complete_reduction=true)
+    ord = G.ord
+    G = collect(G)
+    homogeneityWeight = homogeneity_vector(G)
+
+
+    ###
+    # Starting cone
+    ###
+    C = maximal_groebner_cone(G,ord,homogeneityWeight)
+    workingList = [(G,ord,C,unique_identifying_point(C))] # list of Groebner cones whose neighbours may be unknown
+    finishedList = typeof(workingList)()     # list of Groebner cones whose neighbours are known
+    finishedFacets = Vector{Vector{fmpz}}()  # list of interior facet points whose facet has been traversed
+
+
+    ###
+    # Fan traversal
+    ###
+    while !isempty(workingList)
+        if verbose_level>0
+            println("#workingList / #finishedList: ",length(workingList)," / ",length(finishedList))
+        end
+        # take Groebner cone from workingList
+        G,ord,C,pt = pop!(workingList)
+
+        # enumerate its facets
+        for (u,v) in zip(interior_facet_points(C),outer_normal_vectors(C))
+            # check whether facet is supposed to be traversed
+            # (i.e., if ideal is not quasi-homogeneous,
+            #  check whether facet lies inside the positive orthant)
+            if homogeneityWeight==nothing && !reduce(&,isless.(0,u))
+                continue
+            end
+
+            # check whether facet has already been traversed.
+            i = searchsortedfirst(finishedFacets,u)
+            if i<=length(finishedFacets) && finishedFacets[i]==u
+                continue
+            end
+            insert!(finishedFacets,i,u)
+
+            # compute adjacent Groebner cone
+            Gprime,ordPrime = groebner_flip(G,ord,homogeneityWeight,u,v)
+            Cprime = maximal_groebner_cone(Gprime,ordPrime,homogeneityWeight)
+            ptPrime = unique_identifying_point(Cprime)
+
+            # check whether adjacent cone already in finishedList
+            if insorted((0,0,0,ptPrime),finishedList,by=entry->entry[4])
+                continue
+            end
+
+            # check whether adjacent cone already in workingList
+            i = searchsortedfirst(workingList,(0,0,0,ptPrime),by=entry->entry[4])
+            if i <= length(workingList) && workingList[i][4] == ptPrime
+                continue
+            end
+
+            # insert into workingList in previously computed position
+            insert!(workingList,i,(Gprime,ordPrime,Cprime,ptPrime))
+        end
+
+        # put Groebner cone into finishedList
+        i = searchsortedfirst(finishedList,(0,0,0,pt),by=entry->entry[4])
+        insert!(finishedList,i,(G,ord,C,pt))
+    end
+
+
+    ###
+    # Post processing
+    ###
+    # construct polyhedral fan and return it if nothing else was required
+    Sigma = polyhedral_fan_from_cones([entry[3] for entry in finishedList])
+    if !return_groebner_bases && !return_orderings
+        return Sigma
+    end
+
+    # otherwise return what was required
+    output = Any[]
```

Comment:
Use of `Any` here is redundant.
```suggestion
    output = []
```